spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/ChannelBindingService.java

```diff
@@ -78,10 +76,22 @@ private String removePrefix(String bindingTarget) {
 	}
 
 	public void unbindConsumers(String inputChannelName) {
-		this.binder.unbindConsumers(inputChannelName);
+		if (BinderUtils.isChannelPubSub(this.channelBindingServiceProperties.getBindingDestination(inputChannelName))) {
+			this.binder.unbindPubSubConsumers(inputChannelName, consumerGroup(inputChannelName));
+		}
+		else {
+			this.binder.unbindConsumers(inputChannelName);
+		}
 	}
 
 	public void unbindProducers(String outputChannelName) {
 		this.binder.unbindProducers(outputChannelName);
 	}
+
+	private String consumerGroup(String inputChannelName) {
+		BindingProperties bindingProperties = this.channelBindingServiceProperties.getBindings()
```

Comment:
This will work for now, since it is currently impossible to specify a pubsub scenario without using binding properties. When we remove the distinction between queue/topic, this will have to be updated, since there may be cases when no binding properties are present.